haystack/components/agents/agent.py

```diff
@@ -267,7 +279,7 @@ def run(
         streaming_callback = select_streaming_callback(
             init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=False
         )
-        generator_inputs = self._prepare_generator_inputs(streaming_callback=streaming_callback)
+        generator_inputs = self._prepare_generator_inputs(streaming_callback=streaming_callback, generator_inputs=generation_kwargs)
```

Comment:
I believe this is a typo 
```suggestion
        generator_inputs = self._prepare_generator_inputs(streaming_callback=streaming_callback, generator_kwargs=generation_kwargs)
```